std/encoding.d

```diff
@@ -57,8 +57,11 @@ module std.encoding;
 
 import std.traits;
 import std.range.primitives;
+import std.array;
 
-unittest
+@safe {
```

Comment:
Please let's not go down this path. After much searching I did find where this ended but it's not at all obvious.

I understand the motivation - `@safe:` is impractical the second we need attribute inference *anywhere* in the module. This is something we need to fix in the language, such as with [DIP 79](http://wiki.dlang.org/DIP79) or similar.

Can we please stick to more local attribute applications until we can fix this properly? If I was a maintainer who needed to add a new function to this module and I didn't have previous knowledge of this hack, I'd go crazy.